Add more explicit annotations of top level coexistence (#29921)
akka/akka: akka-docs/src/main/paradox/typed/coexisting.md

@scala[That adds some implicit extension methods that are added to classic and typed `ActorSystem`, `ActorContext` and `ActorRef` in both directions.]
@java[To convert between typed and classic `ActorSystem`, `ActorContext` and `ActorRef` in both directions there are adapter methods in `akka.actor.typed.javadsl.Adapter`.]

> Note the inline comments in the example above. Additionally, note that the above method of using a top level classic actor is the suggested path for this type of co-existence. However, if you prefer to start with a typed top level actor then you can use the implicits and then call `spawn` directly from the typed system:

Reviewer: ```suggestion
Note the inline comments in the example above. 

This method of using a top level classic actor is the suggested path for this type of co-existence. However, if you prefer to start with a typed top level actor then you can use the @scala[implicit `spawn` -method]@java[`Adapter.spawn`] directly from the typed system:
```

(I didn't test render this with paradox though)

Scala
:  @@snip [TypedWatchingClassicSpec.scala](/akka-actor-typed-tests/src/test/scala/docs/akka/typed/coexistence/TypedWatchingClassicSpec.scala) { #create }

Java
:  @@snip [TypedWatchingClassicTest.java](/akka-actor-typed-tests/src/test/java/jdocs/akka/typed/coexistence/TypedWatchingClassicTest.java) { #create }

> Once migrated to a fully typed system the concept is to provide the root actor in the `ActorSystem` constructor, and then spawn actors from the `ActorContext` of the actor itself. The rationale for this is partly about consistency. In a typed system you can’t create children to an arbitrary actor from anywhere in your app without messaging it, so this will also hold true for the User guardian actor. Whereas in a classic system it was mostly not noticed that the guardian was an actor. That noted, in cases where you do need to spawn outside of this guardian then you can use the @ref:[`SpawnProtocol`](./actor-lifecycle.md#spawnprotocol) to spawn as needed.

Reviewer: Perhaps flesh out https://doc.akka.io/docs/akka/current/typed/from-classic.html#actorsystem and link to that instead?

Author: Sorry for the long delay here. I'm unsure to what you are referring. Are you suggesting to put this last paragraph into the from-classic because it is more related to that, or the whole thing? I can understand the former so will move in that direction, but let me know if you meant otherwise.

Author: Moved - and removed "Whereas in a classic system it was mostly not noticed that the guardian was an actor." as it seemed superfluous.
